nixos/modules/services/networking/haproxy.nix

```
@@ -3,12 +3,15 @@
 let
   cfg = config.services.haproxy;
 
-  haproxyCfg = pkgs.writeText "haproxy.conf" ''
+  haproxyCfg = pkgs.runCommand "haproxy.conf" { CONFIG=cfg.config; } ''
+    cat > $out <<EOF
```

Needs to be quoted, otherwise parameter expansion and command substitution will be applied in the heredoc.
```suggestion
    cat > $out <<'EOF'
```